tests/drivers/clock_control/pwm_clock/src/main.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2023 Andriy Gelman <andriy.gelman@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/ztest.h>
+
+#define NODELABEL DT_NODELABEL(samplenode)
+
+ZTEST(pwm_clock, test_pwm_clock_rate_change)
```

Comment:
I would suggest splitting this test in smaller test functions, each testing an API function.